app/src/androidTest/java/ch/hikemate/app/ui/components/ElevationGraphTest.kt

```diff
@@ -0,0 +1,151 @@
+package ch.hikemate.app.ui.components
+
+import android.content.Context
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.unit.dp
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import ch.hikemate.app.R
+import ch.hikemate.app.utils.MapUtils
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class ElevationGraphTest {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    context = ApplicationProvider.getApplicationContext()
+  }
+
+  @Test
+  fun elevationGraph_withValidData_displaysGraph() {
+    val elevationData = listOf(100.0, 200.0, 150.0, 175.0)
+
+    composeTestRule.setContent {
+      ElevationGraph(
+          elevations = elevationData,
+          modifier = Modifier.size(200.dp),
+          styleProperties = ElevationGraphStyleProperties())
+    }
+
+    composeTestRule.waitForIdle()
+
+    // Verify that neither loading nor no data messages are shown
+    composeTestRule
+        .onNode(hasText(context.getString(R.string.elevation_graph_loading_label)))
+        .assertDoesNotExist()
+    composeTestRule
+        .onNode(hasText(context.getString(R.string.hike_card_no_data_label)))
+        .assertDoesNotExist()
```

Comment:
Improvement: Could you also, armed with your knowledge of these pixel tests, please add a test here that shows that the composable contains the hike color? That would be a neat addition that confirms that the hike is displayed in the card.

Reply:
Here it should not be displayed. But I still added the check

Reply:
I'm afraid you've misunderstood, I meant to check wether the color of the hike is on the image, so whether the hike elevation graph is displayed properly.

Reply:
looks great, thanks!